lib/components/resinio/sdk.js

```diff
@@ -134,9 +134,13 @@ exports.createSSHKey = async () => {
   }
 }
 
-exports.removeSSHKeys = () => {
-  return resin.models.key.getAll().each((key) => {
-    return resin.models.key.remove(key.id)
+exports.removeSSHKeyByLabel = (label) => {
+  return resin.models.key.getAll().then((keys) => {
+    resin.models.key.remove(
+      keys.find((elem) => {
+        return elem.title === label
+      }).id
```

Comment:
Keep in mind that `.find()` might not actually find a matching key, and then `.id` will try to access a property of an undefined object.